Merge feat/element-nav with dev

astra137 opened the pull request.

Commit:
feat(router): Stub new components and connect with vue-router

Adds vue-router, stubs new components, and tries to mimick the wireframe. Routing is using params to
select a game and element to edit.

#14

Commit:
build(dependencies): Add vue-router as dependency

Commit:
fix(router): Move game dropdown, simplify element list, prototype element slugs

Adds prototype element slugs that follow game slug and cleans interface a bit.

#8

Commit:
wip: moving feat/app-layout work to another machine

Commit:
Merge dev

Commit:
feat(vuex): Add vuex, tether component props to vuex

Unfinished work towards local editing and element navigation.

re #15

Commit:
feat(vuex): Add new element frame

Unfinished work to allow for changing local state.

Commit:
refactor(vuex): Move some components into folders

Commit:
feat(vuex): Refactor .vue files, revert router to slashless element slugs

Commit:
feat: Change to element-ui; Add createElement

Commit:
feat(element-nav): Add delete element, save to localStorage

Add a dialog for deleting elements and save state to localStorage for easier demo-ing

monochromicon commented:
Looks like the CI build is failing. You need to add the latest version of vue-template-compiler to your dependencies. Try running the build locally and see what happens

Commit:
fix(webpack): Bump vue-template-compiler

Commit:
refactor(package.json): Tweak package.json

Change repository field to shortcut syntax; Reorganize structure.

Commit:
fix(feat/element-nav): Set height of outer router-view

Set height so that vertical scrolling of element nav works as expected.

puregarlic commented on src/games/Game.vue:
```diff
@@ -0,0 +1,80 @@
+  {
+    slug: 'player-rovs',
+    title: 'ROVs',
+    markdown: `# ROVs\n\nThe core game mechanic is driving remotely operated vehicles around a two-dimensional map.\n`
```
How come you're using vanilla strings with `\n` instead of template literals?

puregarlic commented on src/games/Game.vue:
```diff
@@ -0,0 +1,80 @@
+    ElementNav,
+    EditElement
+  },
+  props: ['game', 'slug'],
```
Let's get in the habit of using the more verbose form for props. Sure, it's type checking, but there's almost no effort involved in adding it, and it's useful inline documentation. Information on the syntax can be found [here](http://devdocs.io/vue~2/api/index#props).

puregarlic commented on src/games/elements/EditElement.vue:
```diff
@@ -0,0 +1,33 @@
+<script>
+export default {
+  name: 'EditElement',
+  props: ['markdown'],
```
Same comment on props here.

puregarlic commented on src/games/elements/ElementNav.vue:
```diff
@@ -0,0 +1,63 @@
+
+<template>
+  <el-menu :router="true">
+    <el-dialog
```
Turns out there's another modal-like component that's meant for this purpose: The [Message Box](http://element.eleme.io/#/en-US/component/message-box).

puregarlic commented on src/games/elements/NewElement.vue:
```diff
@@ -0,0 +1,51 @@
+
+export default {
+  name: 'NewElement',
+  props: ['game'],
```
Expand this prop

puregarlic commented on src/games/elements/ElementNav.vue:
```diff
@@ -0,0 +1,63 @@
+<script>
+export default {
+  name: 'ElementNav',
+  props: ['game', 'elements'],
```
Expand these

puregarlic commented on src/App.vue:
```diff
@@ -1,63 +1,86 @@
+      </span>
+    </el-dialog>
+
+    <el-menu mode="horizontal" :router="true">
```
Let's use the prepackaged way to do layouts in Element UI, it's basically made for the look of our application: [Container](http://element.eleme.io/#/en-US/component/container)

puregarlic reviewed (changes requested):
Overall functionality looks great, just needs some minor tweaks before we can fully accept it.

puregarlic committed:
chore(gitattributes): Adds .gitattributes file

Lockfiles suck. With a .gitattributes file, we direct git to consider it as both not text and not
merge-able. Your local editor should still be able to diff it, though.

maccelerated commented:
I'll expand the props, then convert the remove-element feature to message box after that.

Commit:
refactor(props): Expand props to verbose definition

Commit:
feat(ElementNav): Implement MessageBox

Commit:
feat(router): Mimick GitHub style routing

Commit:
refactor(layout): Convert to Element UI layout

#13

Commit:
fix(element-nav): Preserve next element id

Save next element number to localstorage instead of guessing from list length.

re #34

puregarlic reviewed (approved):
Alright, these changes look good for now. Let’s get this merged back to dev.
